v0/util/index.js

```diff
@@ -1546,7 +1546,10 @@ const isOAuthSupported = (destination, destHandler) => {
 
 function isAppleFamily(platform) {
   const appleOsNames = ["ios", "watchos", "ipados", "tvos"];
-  return appleOsNames.includes(platform?.toLowerCase());
+  if (platform) {
+    return appleOsNames.includes(platform?.toLowerCase());
+  }
+  throw new CustomError("Platform is not defined", 400);
```

Comment:
This function is having a side effect the responsibility of validating the existence of platform should be by caller

Reply:
We are checking if platform is string or not.

Reply:
This is an anti-pattern. Since this is just validating the input value's presence, type-checking the input can go in here.

Reply:
The function name is `isAppleFamily` which gives an impression that this function returns a boolean value for a supplied value if it meets the condition of mentioned in function name `verb`

If we would also want to check the validity of the supplied value we should change function name to `validateAppleFamily` a verb which share most context.

Reply:
Yes, any value like "null", "undefined", 2, etc, will fail the apple family check and should return `false`. We shouldn't throw an error. The validation is considered part of the check itself.